tests/spicy/types/unit/field-declaration.spicy

```diff
@@ -0,0 +1,27 @@
+# @TEST-EXEC:  printf 'abcde999abcdef\001\002\003\377' | spicy-driver %INPUT | sort >output
+# @TEST-EXEC:  btest-diff output
+#
+# @TEST-GROUP: spicy-core
+#
+# @TEST-DOC: Test variants of global field declarations.
+
+module Mini;
+
+type FooImpl = unit {
+    data: bytes &size=3;
+};
+
+field X1 = bytes &size=5;
+field X2 = FooImpl &convert=$$.data.to_int();
+field X3 = /a....f/;
+field X4 = int8[3];
+field X5 = uint8 { print "in decl", $$; }
+
+public type Test = unit {
+    x1: X1;
+    x2: X2;
+    x3: X3;
+    x4: X4;
+    x5: X5 { print "in unit", $$; }
+    on %done   { print self; }
+};
```

Comment:
I could see users trying to pass a field decl as a function parameter, so I had a look at the following test:

```.ruby
function foo(x: X1) : uint64 {
    return 0;
}
```

This fails correctly but with a slightly ugly error which one can probably already trigger with the current code:
```
[error] ../tests/spicy/types/unit/field-declaration.spicy:29:14: ID 'X1' does not resolve to a N5hilti11declaration4TypeE (but to unit field)
[error] spicyc: aborting after errors
```

* * *

I also checked whether one can import field decls from other modules, but it looks like this does not yet work:

```.ruby
// a.spicy
module a;

import b;

field I = int8;

type X = unit {
    i1 : I;
    i2 : b::I;
};
```

```.ruby
// b.spicy
module b;

field I = string; // `pub` not recognized here.
```

```
$ spicyc -c a.spicy
[error] a.spicy:9:5: unknown ID 'b::I'
[error] spicyc: aborting after errors
```

Reply:
I'll look into the error message.

Forgot to add a test, but was expecting `public` to work, will give it a try.

Reply:
Confirming that this is a bug: the `public` linkage isn't propagated (and the error message is misleading). I fixed that, however there's now another problem: ID lookups inside the imported module aren't resolved (because we copy the sub-AST over into the importing module). Need to look into how to fix that.